devlib/module/sched.py

```diff
@@ -0,0 +1,262 @@
+#    Copyright 2018 Arm Limited
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import re
+
+from devlib.module import Module
+from devlib.utils.misc import memoized
+
+from past.builtins import basestring
+
+
+class SchedModule(Module):
+
+    name = 'sched'
+
+    sysfs_root = '/sys/devices/system/'
+
+    @staticmethod
+    def probe(target):
```

Comment:
Maybe too generic name? You are probing just for SDs, isn't it?

Reply:
And... since it's not used by this module, I guess it's an API moved here from some other LISA module... hence that's a good chance to have a better descriptive name.

Reply:
For example, such a method should not better be part of the SchedDomainData class?
Something like:
`SchedDomainData.exists()` or `SchedDomainData.valid()` or `SchedDomainData.available()` ?

Reply:
Actually it's imposed by the devlib module API - it's run to see if it makes sense to load the module.